### ISSUE TITLE
ValueError is not supported in `translate_validation`

### ISSUE THREAD

**Issue by anveshagarwal** [carltongibson/django-filter#1145]
https://github.com/carltongibson/django-filter/blob/master/django_filters/utils.py#L319

If `e` is a `ValueError` object, then `e.message` is not a string , it, itself is a `ValueError` object and it is throwing this error 
```  
File "/usr/local/lib/python3.7/site-packages/django_filters/utils.py", line 322, in translate_validation
    for key, error_list in error_dict.as_data().items()
File "/usr/local/lib/python3.7/site-packages/django_filters/utils.py", line 322, in <genexpr>
    for key, error_list in error_dict.as_data().items()
File "/usr/local/lib/python3.7/site-packages/django_filters/utils.py", line 322, in <genexpr>
  File "/usr/local/lib/python3.7/site-packages/django_filters/utils.py", line 321, in <listcomp>
    for e in error_list])
TypeError: unsupported operand type(s) for %: 'ValueError' and 'tuple'
```

**Comment by rpkilby** [comment]
`translate_validation` specifically converts Django's `ValidationError`s to DRF `ValidationError`s. It is not intended to handle other error types. If your code is throwing a `ValueError` during validation, you should catch that and  reraise as a `ValidationError`.

**Comment by anveshagarwal** [comment]
ok. Thanks. But can't it be modified so that it can handle other error types too?

**Comment by rpkilby** [comment]
No, and you wouldn't want to. `ValidationError`s are specifically used by Django forms to indicate a field has failed to validate. Any other exception type would indicate a programming error. If you're getting a `ValueError`, you should determine why and either fix the issue if it's a bug, or determine how best to reraise it as a `ValidationError`.

That said, I'm confused how you're getting a `ValueError` to begin with. The form shouldn't catch/handle it to begin with. What is the value of `filterset.errors`?

**Comment by anveshagarwal** [comment]
actually i am using a custom field that internally uses `dateutil.parser.parse()` function which internally throws `ValueError`. I did the workaround by catching that error and re-raised as `ValidatioError`

**Comment by rpkilby** [comment]
That seems like the correct thing to do. I'm still confused how the `ValueError` was being handled to begin with, as it should have just bubbled to the top.
